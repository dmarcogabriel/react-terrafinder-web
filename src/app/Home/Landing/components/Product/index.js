import React from 'react';
import Button from 'common/components/Button';
import { MdArrowForward } from 'react-icons/md';
import { useHistory } from 'react-router-dom';
import classes from './Product.module.scss';

export default function Product({ image, product }) {
  const history = useHistory();

  const handleSelect = (link) => {
    history.push(link);
  };

  return (
    <div className={classes.container}>
      <div className={classes.product}>
        <img className={classes.image} src={image} alt="test" />

        <div className={classes.content}>
          <p className={classes.title}>{product.title}</p>

          <p className={classes.text}>{product.description}</p>
        </div>
      </div>

      <Button
        type="button"
        dataTestId="actionButton"
        className={classes.actionButton}
        onClick={() => handleSelect(product.link)}
      >
        <p className={classes.buttonText}>{product.buttonText}</p>
        <MdArrowForward />
      </Button>
    </div>
  );
}
