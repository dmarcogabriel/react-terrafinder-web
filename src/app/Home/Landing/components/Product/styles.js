import { styled } from '@mui/material/styles';
import { Button, Box, Card as MuiCard } from '@mui/material';

export const ProductWrapper = styled(Box)(({ theme }) => ({
  position: 'relative',
  width: '100%',
  [theme.breakpoints.up('md')]: {
    width: '30%',
  },
}));

export const Card = styled(MuiCard)({
  display: 'flex',
  flexDirection: 'column',
  height: '305px',
});

export const MoreButton = styled(Button)({
  position: 'absolute',
  bottom: '-1.2rem',
  left: 0,
  right: 0,
  marginLeft: 'auto',
  marginRight: 'auto',
});
