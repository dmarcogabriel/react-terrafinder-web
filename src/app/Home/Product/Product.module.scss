@import '../../../styles/';

.container {
  position: relative;
  margin: 0 $size-lg 2.5rem $size-lg;

  .product {
    border-radius: 0.25rem;
    box-shadow: $shadow-base;
    overflow: hidden;

    .image {
      width: 100%;
    }

    .content {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: $size-md 0 2.5rem;

      .title {
        font-size: 1rem;
        margin-bottom: $size-xs;
      }

      .text {
        font-size: 0.75rem;
        margin: 0 $size-xs;
      }
    }
  }

  .actionButton {
    position: absolute;
    right: 0;
    left: 0;
    margin-left: auto;
    margin-right: auto;
    bottom: -1.5rem;
    display: flex;
    align-items: center;
    justify-content: space-between;

    .buttonText {
      margin-right: 0.5rem;
    }
  }
}
