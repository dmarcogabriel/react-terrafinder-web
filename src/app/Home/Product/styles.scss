@import '../../../styles/';

.container {
  position: relative;
  margin-left: $size-lg;
  margin-right: $size-lg;
  margin-bottom: 2.5rem;
  border-radius: $size-lg;
  box-shadow: $shadow-base;

  .productImage {
    width: 100%;
    border-radius: $size-lg;
  }

  .content {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: $size-md 0;

    .title {
      font-size: $size-md;
      margin-bottom: $size-xs;
    }

    .text {
      font-size: $size-2xs;
      margin: 0 $size-xs;
    }
  }

  .actionButton {
    position: absolute;
    right: 0;
    left: 0;
    margin-left: auto;
    margin-right: auto;
    padding: $size-sm $size-md;
    font-size: $size-md;
    color: $white;
  }
}
