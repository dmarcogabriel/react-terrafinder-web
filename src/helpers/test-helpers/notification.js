export const mockUseNotification = (showNotification = () => {}) => {
  const mockShowNotification = jest.fn(showNotification);

  jest.mock('hooks/useNotification', () => ({
    useNotification: () => ({
      showNotification: mockShowNotification,
    }),
    NOTIFICATION_TYPES: { SUCCESS: 'success', ERROR: 'error' },
  }));

  return { mockShowNotification };
};
