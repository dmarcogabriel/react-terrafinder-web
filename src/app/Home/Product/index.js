import React from 'react';
import './styles.scss';

export default function Product({ image, key }) {
  return (
    <div className="container" key={key}>
      <img className="productImage" src={image} alt="test" />

      <div className="content">
        <p className="title">Anunciar imóvel rural</p>

        <p className="text">
          Quero criar um anúncio para vender minha fazenda, chácara ou sítio.
        </p>
      </div>

      <button className="actionButton" type="button">
        Anunciar
      </button>
    </div>
  );
}
