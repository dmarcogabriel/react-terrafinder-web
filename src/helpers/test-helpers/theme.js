import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

export const withTheme = (component) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
