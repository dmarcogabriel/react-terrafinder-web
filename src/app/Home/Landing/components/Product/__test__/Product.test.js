import React from 'react';
import ReactDOM from 'react-dom';
import { render, fireEvent } from '@testing-library/react';
import { renderWithTheme, withTheme } from 'helpers/test-helpers/theme';
import Product from '..';

const mockFn = jest.fn();
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockFn,
  }),
}));

const mockProduct = {
  image: '',
  product: {
    title: 'test',
    description: 'This is a test',
    buttonText: 'Go to test',
    link: '/test',
  },
};

describe('<Property />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(withTheme(<Product {...mockProduct} />), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should go to link', () => {
    const { getByTestId } = render(withTheme(<Product {...mockProduct} />));

    const actionButton = getByTestId('actionButton');
    fireEvent.click(actionButton);

    expect(mockFn).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const tree = renderWithTheme(<Product {...mockProduct} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
