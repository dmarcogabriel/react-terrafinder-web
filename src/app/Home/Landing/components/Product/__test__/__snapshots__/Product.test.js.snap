// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Property /> matches snapshot 1`] = `
.c0 {
  cursor: pointer;
  text-align: center;
  font-family: Roboto;
  box-shadow: 0 0.2rem 0.6rem 0 rgba(0,0,0,0.7);
  padding: 0.75rem;
  border-radius: 0.25rem;
  background: #fff;
  color: #333;
}

<div>
  <div>
    <img
      alt="test"
      src=""
    />
    <div>
      <p>
        test
      </p>
      <p>
        This is a test
      </p>
    </div>
  </div>
  <button
    className="c0"
    data-testid="actionButton"
    onClick={[Function]}
    type="button"
  >
    <p>
      Go to test
    </p>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </button>
</div>
`;
